demisto_sdk/commands/test_content/xsoar_tools/xsoar_client.py

```diff
@@ -0,0 +1,697 @@
+    def run_cli_command(
+        self,
+        command: str,
+        investigation_id: Optional[str] = None,
+        response_type: str = "object",
+    ):
+
+        update_entry = {
+            "investigationId": investigation_id,
+            "data": "!DeleteContext all=yes",
+        }
+
+        self.client.investigation_add_entries_sync(update_entry=update_entry)
+
+        update_entry = {"investigationId": investigation_id, "data": command}
+        response = self.client.investigation_add_entries_sync(update_entry=update_entry)
+
+        context = self.get_investigation_context(investigation_id)
+        return response, context
+
+    @retry_http_request()
+    def get_investigation_context(
+        self, investigation_id: str, response_type: str = "object"
+    ):
+        raw_response, _, _ = demisto_client.generic_request_func(
+            self=self.client,
+            method="POST",
+            path=f"/investigation/{investigation_id}/context",
+            response_type=response_type,
+            body={"query": "${.}"},
+        )
+        return raw_response
+
+    @retry_http_request()
+    def delete_playbook(self, name: str, _id: str, response_type: str = "object"):
+        raw_response, _, _ = demisto_client.generic_request_func(
+            self=self.client,
+            method="POST",
+            path="/playbook/delete",
+            response_type=response_type,
+            body={"id": _id, "name": name},
+        )
+        return raw_response
+
+    @retry_http_request(times=20, delay=3)
+    def get_playbook_state(self, incident_id: str, response_type: str = "object"):
+        raw_response, _, _ = demisto_client.generic_request_func(
+            self=self.client,
+            method="GET",
+            path=f"/inv-playbook/{incident_id}",
+            response_type=response_type,
+        )
+        return raw_response
+
+    @retry_http_request()
+    def get_investigation_status(self, incident_id: str):
+        raw_response, status_code, _ = demisto_client.generic_request_func(
+            self=self.client,
+            method="POST",
+            path=f"/investigation/{urllib.parse.quote(incident_id)}",
+            body={"pageSize": 1000},
+            response_type="object",
+        )
+        if status_code == 200:
+            return raw_response
+        raise ApiException(
+            status=status_code,
+            reason=f"Failed to get investigation status for {incident_id=}, {raw_response=}",
+        )
+
+    @retry_http_request()
+    def get_playbook_failure_by_incident(self, incident_id: str):
+        investigation_status = self.get_investigation_status(incident_id)
+        entries = investigation_status["entries"]
+        error_entries = {}
+        for entry in entries:
+            if entry["type"] == self.ENTRY_TYPE_ERROR and entry["parentContent"]:
+                # Checks for passwords and replaces them with "******"
+                parent_content = re.sub(
```

Comment:
why would we get passwords from the api? We need to make sure nothing from this response is printed

Reply:
it is possible that once a task fails in a playbook, that it will print secrets from the stderr.
that just verifies that it won't be logged into the SDK in any way.